app/cmd/preview.go

```diff
@@ -375,6 +411,39 @@ func collectLinkPaths(target string) ([]string, error) {
 	return m.Links, nil
 }
 
+// collectDataPaths takes a target, reads it, and scans the file for data paths and collects them
+func collectDataPaths(target string) ([]string, error) {
```

Comment:
This plus collectLinkPaths means that we're reading the file twice, which shouldn't be too much of a performance hit since almost all the bottleneck is on the wire. The other thought would be to do the sql file parsing in the same location where we do the link extraction, `mdlinkparser/md_link_parser.go`

Tbh though writing a solution for it there would probably be less clear than how this is doing it.